cms6 upgrade

ishannz opened the pull request:
<!--
  Thanks for contributing, you're awesome! ⭐

  Please read https://docs.silverstripe.org/en/contributing/code/ if you haven't contributed to this project recently.
-->
## Description
<!--
  Please describe expected and observed behaviour, and what you're fixing.
  For visual fixes, please include tested browsers and screenshots.
-->

## Manual testing steps
<!--
  Include any manual testing steps here which a reviewer can perform to validate your pull request works correctly.
  Note that this DOES NOT replace unit or end-to-end tests.
-->

## Issues
<!--
  List all issues here that this pull request fixes/resolves.
  If there is no issue already, create a new one! You must link your pull request to at least one issue.
-->
- #

## Pull request checklist
<!--
  PLEASE check each of these to ensure you have done everything you need to do!
  If there's something in this list you need help with, please ask so that we can assist you.
-->
- [ ] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [ ] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [ ] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [ ] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [ ] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [ ] This change is covered with tests (or tests aren't necessary for this change)
- [ ] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [ ] CI is green

ishannz committed:
cms6 upgrade

satrun77 committed:
Remove term extension

The Pages field is no longer gridfield

lianna-blanca committed:
CMS6 compatibility updates

lianna-blanca committed:
Merge pull request #1 from silverstripeltd/pulls/cms6-fixes

CMS6 compatibility updates

lianna-blanca committed:
Fix hide_pagetypes config for CMS 6 — move from subclass PHP to SiteTree YAML

lianna-blanca committed:
Merge pull request #2 from silverstripeltd/cms6/hide-pagetypes-config

CMS 6: Fix hide_pagetypes config

michalkleiner commented on composer.json:
```diff
@@ -8,24 +8,30 @@
     "suggest": {
         "silverstripe/crontask": "Allows CWP cleanup tasks to be scheduled"
     },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/silverstripeltd/cwp-core.git"
+        }
+    ],
```
Just making a note that this needs to be removed before merging.

michalkleiner replied:
Might even remove it now and add instructions to the PR readme that this fork with the cwp-core repo is needed for testing. We don't want this accidentally merged in.

michalkleiner commented on _config/config.yml:
```diff
@@ -9,10 +9,11 @@ SilverStripe\SiteConfig\SiteConfig:
   extensions:
     - CWP\CWP\Extensions\CwpSiteTreeExtension
-
-SilverStripe\Taxonomy\TaxonomyTerm:
-  extensions:
-   - CWP\CWP\Extensions\TaxonomyTermExtension
```
Why is this being removed? I guess we can afford removing it in a new major version, but it might come as a surprise to some users/devs and could be missed that it needs to be replaced at project level.

edwilde committed:
fix(EventPage): remove SS6 auto-scaffolded Date/StartTime/EndTime before adding custom FieldGroup

SS6 auto-scaffolds $db Boolean/Date/Time/etc fields into getCMSFields() output.
EventPage builds a custom 'Date and time' FieldGroup containing Date + StartTime +
EndTime fields; without removing the scaffolded versions first, SS6 errors with
'a field called StartTime appears twice' on first CMS render.

Matches the pattern already used for Date (line was removeByName('Date') only).
Extended to ['Date', 'StartTime', 'EndTime'].
